code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm

```diff
@@ -304,7 +304,7 @@
 /datum/reagent/medicine/charcoal/on_mob_life(mob/living/M)
 	M.adjustToxLoss(-2*REM)
 	for(var/datum/reagent/R in M.reagents.reagent_list)
-		if(R != src)
+		if(R != src && !istype(R, /datum/reagent/toxin/wasting_toxin)) //Changeling death sting toxin can't be removed by charcoal, only delayed
```

Comment:
Maybe you could change charcoal to only purge datum/reagent/toxin children instead of all reagent and make the ling reagent into a reagent that isn't categorised as toxin?